lib/mayaUsd/ufe/UsdAttributes.cpp

```diff
@@ -454,8 +460,94 @@ bool UsdAttributes::doRemoveAttribute(const UsdSceneItem::Ptr& item, const std::
     }
     return false;
 }
+bool UsdAttributes::canRenameAttribute(
+    const UsdSceneItem::Ptr& sceneItem,
+    const std::string&       targetName,
+    const std::string&       newName)
+{
+    if (!sceneItem || !sceneItem->prim().IsActive()
+        || !UsdAttributes(sceneItem).hasAttribute(targetName)) {
+        return false;
+    }
+
+    // Do not rename if the newName is same as the targetName
+    if (targetName == newName) {
+        return false;
+    }
+
+    // Do not rename if an attribute already exists with the newName
+    if (UsdAttributes(sceneItem).hasAttribute(newName)) {
+        return false;
+    }
+
+    // We can rename attributes if they are:
+    // or boundary attributes,
+    // or custom attributes and not boundary
+
+    PXR_NS::TfToken nameAsToken(targetName);
+    auto            prim = sceneItem->prim();
+    auto            attribute = prim.GetAttribute(nameAsToken);
+
+    PXR_NS::UsdShadeNodeGraph      ngPrim(prim);
+    PXR_NS::UsdShadeConnectableAPI connectApi(prim);
+    if (ngPrim && connectApi) {
+        auto baseNameAndType = PXR_NS::UsdShadeUtils::GetBaseNameAndType(nameAsToken);
+        if (baseNameAndType.second == PXR_NS::UsdShadeAttributeType::Output) {
+            PXR_NS::UsdShadeMaterial matPrim(prim);
+
+            if (matPrim) {
+                // Can not rename the 3 main material outputs as they are part of the schema
+                if (baseNameAndType.first == PXR_NS::UsdShadeTokens->surface
+                    || baseNameAndType.first == PXR_NS::UsdShadeTokens->displacement
+                    || baseNameAndType.first == PXR_NS::UsdShadeTokens->volume) {
+                    return false;
+                }
+            }
+
+            for (auto&& authoredOutput : connectApi.GetOutputs(true)) {
+                if (authoredOutput.GetFullName() == targetName) {
+                    return true;
+                }
+            }
+        } else if (baseNameAndType.second == PXR_NS::UsdShadeAttributeType::Input) {
+            for (auto&& authoredInput : connectApi.GetInputs(true)) {
+                if (authoredInput.GetFullName() == targetName) {
+                    return true;
+                }
+            }
+        }
+    }
+
+    // Custom attribute AND not boundary
+    if (attribute.IsCustom()) {
+        return true;
+    }
+
+    return false;
+}
+
+bool UsdAttributes::doRenameAttribute(
+    const UsdSceneItem::Ptr& sceneItem,
+    const std::string&       targetName,
+    const std::string&       newName)
+{
+    // Avoid checks since we have already did them
+    PXR_NS::TfToken           nameAsToken(targetName);
+    auto                      prim = sceneItem->prim();
+    auto                      attribute = prim.GetAttribute(nameAsToken);
+    PXR_NS::UsdShadeNodeGraph ngPrim(prim);
```

Comment:
This unused variable will most probably cause the preflight to fail on Linux and OS/X.

Reply:
I removed it also in the new version since I don't use it.